crates/bevy_reflect/src/reflect.rs

```diff
@@ -106,32 +106,34 @@ where
     /// Casts this type to a boxed, reflected value.
     ///
     /// This is useful for coercing trait objects.
-    fn into_partial_reflect(self: Box<Self>) -> Box<dyn PartialReflect>;
+    fn into_partial_reflect(self: Box<Self>) -> Box<dyn PartialReflect + Send + Sync>;
```

Comment:
Hm, I feel like this sort of defeats the purpose, no?

The only way we can return `Box::new(self)` here is if `Self: Send + Sync`.